pkg/controllers/awsloadbalancercontroller/controller.go

```diff
@@ -79,49 +80,65 @@ type AWSLoadBalancerControllerReconciler struct {
 func (r *AWSLoadBalancerControllerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	_ = log.FromContext(ctx)
 
-	lbController := &albo.AWSLoadBalancerController{}
-	if err := r.Client.Get(ctx, req.NamespacedName, lbController); err != nil {
-		if errors.IsNotFound(err) {
-			return ctrl.Result{}, nil
-		}
-		return ctrl.Result{}, fmt.Errorf("failed to get AWSLoadBalancerController %s: %w", req, err)
+	lbController, exists, err := r.getAWSLoadBalancerController(ctx, req.Name)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get AWSLoadBalancerController %s: %w", req.Name, err)
+	}
+	if !exists {
+		return ctrl.Result{}, nil
 	}
 
 	servingSecretName := fmt.Sprintf("%s-serving-%s", controllerResourcePrefix, lbController.Name)
 
 	// if the processed subnets have not yet been written into the status or if the tagging policy has changed then update the subnets
 	if lbController.Status.Subnets == nil || (lbController.Spec.SubnetTagging != lbController.Status.Subnets.SubnetTagging) {
-		err := r.tagSubnets(ctx, lbController)
+		internalSubnets, publicSubnets, taggedSubnets, untaggedSubnets, err := r.tagSubnets(ctx, lbController)
 		if err != nil {
 			return ctrl.Result{}, fmt.Errorf("failed to update subnets: %w", err)
 		}
+		err = r.updateStatusSubnets(ctx, lbController, internalSubnets, publicSubnets, untaggedSubnets, taggedSubnets, lbController.Spec.SubnetTagging)
```

Comment:
I've split the updates into multiple stages so that the user gets some feedback through the status. Say if the deployment cannot be created due to a bad `CredentialsRequest` then at least that shows up in the status.

Reply:
Understood, It's just that reloading the controller multiple times within a single reconcile event seems to increase chaces of failure. Can we change `r.Status().Update()` to a normal k8s client Update() call instead? WDYT?